src/main/java/htsjdk/samtools/BAMIndexMetaData.java

```diff
@@ -149,27 +149,23 @@ void recordMetaData(final SAMRecord rec) {
     /**
      * @param slice
      */
-    void recordMetaData(Slice slice) {
-
-        final int alignmentStart = slice.alignmentStart;
-        if (alignmentStart == SAMRecord.NO_ALIGNMENT_START) {
-            noCoordinateRecords+=slice.nofRecords;
+    void recordMetaData(final Slice slice) {
+         if (slice.isUnmapped()) {
+            noCoordinateRecords += slice.unplacedReadsCount;
             return;
         }
+        else {
+            alignedRecords += slice.mappedReadsCount;
+            unAlignedRecords += slice.unmappedReadsCount;
+        }
 
         final long start = slice.offset;
-        final long end = slice.offset + 0;
 
-        if (slice.alignmentSpan < 1) {

```

Comment:
This was just wrong.  It's necessary to check the unmapped flag per-record.  (buildSlice now does this via updateRecordCounts)